### PR TITLE
feat(admin): expose detailed client quota mutation outcomes

### LINKED ISSUE #3134: feat(admin): return detailed client-quota mutation outcomes
Child of #3126; depends on the shared outcome contracts in #3130.

Implement detailed client-quota alteration outcomes keyed by the complete quota entity identity, including nullable/default entity components. Preserve validate-only semantics, mixed entity errors and identity equality.

Acceptance requirements inherited from #3126:
- Add capability interfaces/extensions without breaking custom IAdminClient implementations. Keep existing convenience methods and exception behavior compatible, sharing implementation where practical.
- Retain every confirmed success/failure and original broker error code/message. Distinguish unknown outcomes after transport failure or cancellation from confirmed outcomes; missing response entries never mean success.
- Retry only unresolved/retriable work when protocol semantics permit. Preserve protection against ambiguous non-idempotent retries (#1341).
- Implement matching Dekaf.Testing behavior, public API baselines, documentation with targeted-retry examples, and the parent coverage matrix.
- Add TUnit tests for validation/empty input, cancellation/timeouts, authorization, routing changes where applicable, mixed outcomes, missing responses and ambiguous transport failure. Add appropriate Kafka integration coverage.
- Supply exact-head hosted performance evidence under CLAUDE.md. Do not add work to producer/consumer per-message paths.

Preserve existing detailed APIs, including member removal, feature updates and Streams group offsets. See the [Kafka Admin result conventions](https://kafka.apache.org/43/javadoc/org/apache/kafka/clients/admin/Admin.html).

### PR DESCRIPTION
**Pull request by thomhurst**
Client quota batches can succeed for some entities while other entities fail. Add `AlterClientQuotasDetailedAsync` through an optional capability and `IAdminClient` extension, returning each entity's confirmed or uncertain outcome without replaying successful siblings. Existing convenience APIs and custom `IAdminClient` implementations remain compatible.

Keys retain complete, order-independent quota identity, including null/default and empty-string components. Inputs are snapshotted before asynchronous work; duplicate identities, component types and operation keys are rejected. Validate-only semantics and broker errors are preserved, and discovery, sends and safe retries share the operation deadline. Matching simulator operations, public API entries and documentation cover the same outcomes. The simulator retains its existing limitation: it does not emulate every broker quota configuration rule; fault injection models broker rejections.

Closes #3134

Validation at e347ca1e750b0e489d71b0e3d79090f6aa2c3485, rebased onto main 898c9432b9658ed39c18a9a90747df7c68a6ed75: 132 focused detailed-admin/simulator unit tests pass on each of net10.0 and net8.0; six Kafka quota/security/share-group integration tests pass; all 37 performance-selector tests, documentation production build, repository artifact policy and whitespace checks pass. Rebase conflicts preserve both quota and security APIs, documentation and benchmark mappings. Quota implementations and both quota benchmark fixtures are unchanged from 19a406965ec4f624ccf07cebe4cc9cab2c4023e2.

Performance: two maintained steady-state MemoryDiagnoser fixtures cover broker-facing result processing and direct in-memory mutations. Each has four expanded cases using one/32 entities. The simulator fixture also has ThreadingDiagnoser and synchronized concurrent callers; those measurements include Task.Run/barrier/Task.WhenAll orchestration and do not claim an isolated lock cost. Both fixtures passed local Short validation at their previously measured revisions. Costs are per administrative call/entity; no producer/consumer per-message work is added. No speedup or protected-metric tradeoff is claimed. The new APIs are absent from the baseline, so new fixtures can be candidate-only NOT COMPARED under the gate. All 40 checks pass on this head. Hosted performance gate https://github.com/thomhurst/Dekaf/actions/runs/34557518962 passes; CI https://github.com/thomhurst/Dekaf/actions/runs/34557519042 passes. Current-head Claude review is CLEAR, Greptile reports 5/5, and both review threads are resolved. This administrative routing change does not alter send/receive loops, connection lifecycle or stress-lane behavior and requires no paid stress run.

Current source, raw validation logs, reports and loaded test binaries are retained outside removable worktrees at C:/git/Dekaf-evidence/pr-3238/e347ca1e750b0e489d71b0e3d79090f6aa2c3485 with 1,650 SHA-256 inventory entries. Copies were compared against their originals before cleanup. Earlier broker-facing fixture and feature validation remain at C:/git/Dekaf-evidence/pr-3238/47271636d744d12284daf931de2689ea4d0643c2; the final simulator Short exports, generated drivers and loaded binaries remain at C:/git/Dekaf-evidence/pr-3238/19a406965ec4f624ccf07cebe4cc9cab2c4023e2 with 1,800 inventory entries. Earlier measurements retain their original revision attribution.

Before merge, all SHA-256 inventories were reverified: 1,650 files at the current head, 1,800 at the simulator measurement revision, and 2,650 at the original feature/benchmark revision. Original-copy comparisons were completed before their worktrees were removed. Repository rules require CI Passed and Dependency Review and do not require a human approval.

### CODE REVIEW

**Comment by thomhurst** [comment]
Rebased onto main 16991110b and addressed the benchmark-selection review finding in bc18194cf1a9aad49711c53ef8c559eed6c426ba. The Admin component mapping now includes AdminDetailedClientQuotaBenchmarks alongside the existing detailed-topic and member-removal fixtures. A selector regression test verifies both quota implementation and shared mutation-runner changes select it.

Current source passes 144 focused quota, shared-mutation, and member-removal tests on each of net10.0 and net8.0, plus 31 performance-tooling tests. This update changes the selection mapping and resolves a public-API baseline insertion conflict; quota implementation and its benchmark fixture are unchanged. Earlier integration/documentation/benchmark validation is retained at the prior measured SHA and was not repeated. Fresh-head CI and hosted acceptance are pending.

Rebase logs remain at C:/git/Dekaf-evidence/pr-3238/rebase-20260911. Source, test reports, and loaded unit-test binaries are copied with 1,296 SHA-256-verified files at C:/git/Dekaf-evidence/pr-3238/bc18194cf1a9aad49711c53ef8c559eed6c426ba/inventory.json.

File: tools/Dekaf.Benchmarks/Benchmarks/Unit/AdminDetailedClientQuotaBenchmarks.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+using Dekaf.Metadata;
+using Dekaf.Networking;
+using Dekaf.Protocol;
+using Dekaf.Protocol.Messages;
+
+namespace Dekaf.Benchmarks.Benchmarks.Unit;
+
+/// <summary>Steady-state per-call quota snapshots, request creation and complete-entity result correlation.</summary>
+[MemoryDiagnoser]
+public class AdminDetailedClientQuotaBenchmarks
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/Dekaf.Benchmarks/Benchmarks/Unit/AdminDetailedClientQuotaBenchmarks.cs:12]: Fixed in bc18194cf1a9aad49711c53ef8c559eed6c426ba. The src/Dekaf/Admin/ component mapping now includes AdminDetailedClientQuotaBenchmarks. A selector test verifies quota and shared mutation-runner source changes retain this coverage. All 31 tooling tests and 144 focused unit cases on each framework pass; fresh-head CI is pending.

**Comment by thomhurst** [comment]
Rebased onto main a83fb78cf to include #3237's CLI initialization fix. The quota benchmark job in run 34549243221 failed during dotnet new with NuGet-Migrations/EEXIST, before any measurements. Original artifacts and the failed job log are retained under C:/git/Dekaf-evidence/pr-3238/bc18194cf1a9aad49711c53ef8c559eed6c426ba/gate-34549243221-quota and C:/git/Dekaf-evidence/pr-3238/gate-34549243221-failure.log. No rerun was requested.

The product patch is unchanged (git range-diff reports equality). All 34 tooling tests pass on new head 8f372c20f926c12f757b5cfbabdeb3b6173f66b3. Current source and raw tooling output have a SHA-256 inventory at C:/git/Dekaf-evidence/pr-3238/8f372c20f926c12f757b5cfbabdeb3b6173f66b3; prior product evidence retains its original revision attribution. Fresh CI/review is pending.

**Comment by thomhurst** [comment]
Rebased onto main ba80bf245 after #3239 merged. Resolved the shared mutation-runner conflict by preserving coordinator discovery for share-group operations and broker-or-controller selection for quotas, and updated quotas to the renamed ExecuteDetailedMutationAsync helper. Both benchmark mappings and public API additions remain present.

Current head fcb89539f8b8c591cd95212967f13d241247237d passes 113 focused quota/shared-mutation/share-group tests on each of net10.0 and net8.0, all three Kafka quota/share-group integration cases, 35 tooling tests, and repository artifact/whitespace checks. Existing benchmark fixtures are unchanged; earlier benchmark measurements retain their original revision attribution. Fresh hosted CI/review is pending.

Source, logs, reports and loaded unit/integration binaries are retained with 1,634 SHA-256-verified files at C:/git/Dekaf-evidence/pr-3238/fcb89539f8b8c591cd95212967f13d241247237d. Copied files were verified against their worktree originals before cleanup.

File: src/Dekaf.Testing/InMemoryAdminClient.DetailedClientQuotaMutations.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+using Dekaf.Protocol;
+
+namespace Dekaf.Testing;
+
+public sealed partial class InMemoryAdminClient : IDetailedClientQuotaMutationAdminClient
+{
+    /// <inheritdoc />
+    public ValueTask<IReadOnlyDictionary<ClientQuotaEntity, AdminMutationResult>> AlterClientQuotasDetailedAsync(
+        IEnumerable<ClientQuotaAlteration> alterations, AlterClientQuotasOptions? options = null,
+        CancellationToken cancellationToken = default)
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply src/Dekaf.Testing/InMemoryAdminClient.DetailedClientQuotaMutations.cs:11]: Addressed in 19a406965ec4f624ccf07cebe4cc9cab2c4023e2. Added InMemoryDetailedClientQuotaBenchmarks with four steady-state MemoryDiagnoser/ThreadingDiagnoser cases: one/32 retained entities, either one call or two synchronized callers sharing the same InMemoryAdminClient and quota lock. Setup verifies all outcomes and stored values. The concurrent measurement includes Task.Run, the start barrier, both calls and Task.WhenAll; its contention column includes orchestration, so this is not claimed as an isolated lock cost.

All four final Short cases pass. Diagnostic means are 289 ns/1.5 KB and 5.70 us/27.95 KB for single calls, and 1.87 us/3.6 KB and 17.96 us/56.57 KB for concurrent pairs. No speedup or protected-metric tradeoff is claimed. The earlier Dry fixture also passed four cases before a small return-value/validation simplification. The permanent src/Dekaf.Testing/ mapping and selector regression test retain this workload on future simulator changes; 36 tooling tests pass. Product code is unchanged from the prior green head.

Final source, raw exports/logs and generated benchmark drivers/binaries are retained in C:/git/Dekaf-evidence/pr-3238/19a406965ec4f624ccf07cebe4cc9cab2c4023e2 with a 1,800-file SHA-256 inventory. Copied benchmark binaries were verified against their originals before cleanup. Fresh CI/review is pending; this thread remains open for the subsequent review cycle.

**Comment by thomhurst** [comment]
Rebased onto main 898c9432b at e347ca1e750b0e489d71b0e3d79090f6aa2c3485. Both quota implementations and both benchmark fixtures are unchanged. Conflicts retain the newly landed security APIs and benchmark mapping alongside quota coverage.

132 focused unit tests pass on each framework, six Kafka integration tests pass, and all 37 selector tests, docs build, artifact-policy and whitespace checks pass. Source, raw logs, reports and loaded binaries have 1,650 verified SHA-256 inventory entries at C:/git/Dekaf-evidence/pr-3238/e347ca1e750b0e489d71b0e3d79090f6aa2c3485. Previous benchmark evidence remains attributed to its measured SHA. Fresh CI/performance and review are pending.
